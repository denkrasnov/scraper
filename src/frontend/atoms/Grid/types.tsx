import { ReactNode } from "react";

export interface GridProps {
  children: ReactNode;
  gridGap?: string;
  gridTemplateColumns?: string;
}
