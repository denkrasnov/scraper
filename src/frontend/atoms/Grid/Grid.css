.root {
  display: grid;
}

.gridProducts {
  grid-gap: 25px;
  grid-template-columns: repeat(auto-fit, minmax(280px, 1fr));
}

.result {
  grid-template-columns: auto;
  grid-template-areas:
    "filter"
    "result";
  height: 100%;
}

.main {
  grid-template-columns: 1fr;
  grid-template-rows: auto 1fr;
  grid-template-areas:
    "header"
    "banner"
    "body";
}

@media (min-width: 992px) {
  .result {
    grid-gap: 25px;
    grid-template-columns: minmax(200px, auto) minmax(auto, 1000px) minmax(
        200px,
        auto
      );
    grid-template-areas: "filter result ads";
  }
}
