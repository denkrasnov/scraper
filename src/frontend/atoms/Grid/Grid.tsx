import React, { FC } from "react";

import { GridProps } from "./types";
import styles from "./Grid.css";

const Grid: FC<GridProps> = props => {
  const { children, ...rest } = props;
  return (
    <div className={styles.Grid} style={{ ...rest }}>
      {children}
    </div>
  );
};

export default Grid;
