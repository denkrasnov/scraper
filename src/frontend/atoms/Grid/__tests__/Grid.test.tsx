import React from "react";
import { shallow } from "enzyme";

import Grid from "../Grid";

describe("Grid", () => {
  const children = <div />;

  it("should render", () => {
    const component = shallow(<Grid>{children}</Grid>);

    expect(component).toMatchSnapshot();
  });
});
