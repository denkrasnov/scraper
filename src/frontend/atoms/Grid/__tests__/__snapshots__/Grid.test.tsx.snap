// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid should render 1`] = `
<div
  className="Grid"
  style={Object {}}
>
  <div />
</div>
`;
